pkgs/development/libraries/qt-6/cmake.patch

```diff
@@ -0,0 +1,129 @@
+commit 4f497c358e0386b65df1c1d636aadf72f8647134
+Author: Nick Cao <nickcao@nichi.co>
+Date:   Tue Apr 19 13:49:59 2022 +0800
+
+    patch cmake file generation for nixpkgs packaging
+    
+    As of qt 6.3.0, installing components into different prefixes is not
+    supported. To workaround that, we move files to their designated in the
+    postInstall hook. However the generated cmake files still have
+    references to the original prefix, and would cause issues when using
+    said components as the dependency of other packages. The purpose of this
+    patch is to closely match the output layout of qt, and rewrite the
+    generated cmake files to point to the corrected pathes.
```

Comment:
Why isn't this possible with `qt.conf`?

Reply:
qt.conf is used only by qmake, not by cmake (?)

the default qt.conf file is `dirname(qmake)/qt.conf` and qmake is part of qtbase
since other qt modules depend on qtbase, we would need to generate dynamic qt.conf files

i ignored this problem, since qmake is deprecated in favor of cmake